### PR TITLE
Remove deprecated codes

### PR DESCRIPTION
**Pull request by ji-huazhong**
# What does this PR do?

This PR removes some deprecated code:
- remove `xpu_backend` training argument, for it is deprecated and will be remove in version 4.31 of Transformers.
- remove some codes that will never be executed.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@sgugger

### CODE REVIEW

**Comment by statelesshz** [comment]
test failed, but it doesn't seem to be caused by my modification

**Review by sgugger — APPROVED** [review]
Thanks for your PR!
LGTM, but @muellerzr and @pacman100 could you double-check the grad scaler part (we don't need it anymore in most cases since it is in Accelerate now, but want to make sure!)

File: src/transformers/trainer.py [hunk 1]
```diff
@@ -629,20 +629,7 @@ def __init__(
...
-                        elif self.fsdp is not None:
-                            from torch.distributed.fsdp.sharded_grad_scaler import (
-                                ShardedGradScaler as FSDPShardedGradScaler,
-                            )
-
-                            self.scaler = FSDPShardedGradScaler()
-                        elif is_torch_tpu_available():
-                            from torch_xla.amp import GradScaler
-
-                            self.scaler = GradScaler()
```
Comment (muellerzr) [inline src/transformers/trainer.py:643]: We can't quite get rid of this yet, as we don't have it in Accelerate.

Comment (muellerzr) [reply src/transformers/trainer.py:643]: Let me check something first real quick

Comment (muellerzr) [reply src/transformers/trainer.py:643]: Actually, this logic was always wrong to begin with

**Review by muellerzr — APPROVED** [review]
Thanks!

**Pull request merged**